File: src/main/java/com/sedmelluq/discord/lavaplayer/track/playback/TerminatorAudioFrame.java
package com.sedmelluq.discord.lavaplayer.track.playback;

import com.sedmelluq.discord.lavaplayer.format.AudioDataFormat;

/**
 * Audio frame where {@link #isTerminator()} is <code>true</code>.
 */
public class TerminatorAudioFrame implements AudioFrame {
  public static final TerminatorAudioFrame INSTANCE = new TerminatorAudioFrame();

  @Override
  public long getTimecode() {
    throw new UnsupportedOperationException();
  }

  @Override
  public int getVolume() {
    throw new UnsupportedOperationException();
  }

  @Override
  public int getDataLength() {
    throw new UnsupportedOperationException();
  }

  @Override
  public byte[] getData() {
    throw new UnsupportedOperationException();
  }

  @Override
  public void getData(byte[] buffer, int offset) {
    throw new UnsupportedOperationException();
  }

  @Override
  public AudioDataFormat getFormat() {
    throw new UnsupportedOperationException();
  }

  @Override
  public boolean isTerminator() {
    return true;
  }
}

File: main/src/main/java/com/sedmelluq/discord/lavaplayer/track/playback/TerminatorAudioFrame.kt
package com.sedmelluq.discord.lavaplayer.track.playback

import com.sedmelluq.discord.lavaplayer.format.AudioDataFormat

/**
 * Audio frame where [isTerminator] is `true`.
 */
open class TerminatorAudioFrame : AudioFrame {
    companion object INSTANCE : TerminatorAudioFrame()

    override val timecode: Long
        get() = throw UnsupportedOperationException()

    override val volume: Int
        get() = throw UnsupportedOperationException()

    override val dataLength: Int
        get() = throw UnsupportedOperationException()

    override val data: ByteArray
        get() = throw UnsupportedOperationException()

    override val format: AudioDataFormat
        get() = throw UnsupportedOperationException()

    override val isTerminator: Boolean
        get() = true

    override fun getData(buffer: ByteArray, offset: Int) =
        throw UnsupportedOperationException()
}
